iotfunctions/util.py

```diff
@@ -421,13 +421,41 @@ def get_fn_expression_args(function_metadata, kpi_metadata):
     args = kpi_metadata.get('input', {})
 
     for (arg, value) in list(args.items()):
-        if arg == 'expression':
+        if arg == 'expression' and value is not None:
             expressions.append(value)
             logger.debug('Found expression %s', value)
 
     return infer_data_items(expressions)
 
 
+def get_fn_scope_sources(scope_key, kpi):
+    scope_sources = set()
+    if kpi.get(scope_key):
+        scope = kpi.get(scope_key)
+        if scope.get('type') == 'DIMENSIONS':
+            for dimension_filter in scope.get('dimensions'):
+                dimension_name = dimension_filter['dimension_name']
```

Comment:
same as above, name and value property instead of dimension_name and dimension_value

Reply:
@herleraja Done